src/Core/Framework/Plugin/PluginLifecycleService.php

```diff
@@ -508,4 +528,25 @@ private function getPluginInstance(string $pluginBaseClassString): Plugin
 
         return $this->getPluginBaseClass($pluginBaseClassString);
     }
+
+    private function savePluginConfiguration(Plugin $plugin, InstallContext $context): void
+    {
+        $config     = $this->configReader->getConfigFromBundle($plugin);
+        $prefix     = $plugin->getName() . '.config.';
```

Comment:
please consider that this is not necessarily always `config`. See https://github.com/shopware/platform/blob/master/src/Core/System/SystemConfig/Util/ConfigReader.php#L21

Reply:
How can I use custom config names it my plugin? I can't find such information in the documentation - therefore I can't figure out how to scan for available config names without reading the content of every possible config file.

I have already looked into the code and found out that the domain (which can define the custom config name) will only use "config" for plugins, [see here](https://github.com/shopware/platform/blob/master/src/Administration/Resources/app/administration/src/module/sw-plugin/component/sw-plugin-config/index.js#L13).

Reply:
you can name the file like you want. like you can see in the ConfigReader you could provide another name for your config file and this will be read.

the default is `config` and that's why the admin component uses this. 

but you could reading from different files. you only have to adjust your search for the config: https://github.com/shopwareLabs/SwagPayPal/blob/master/src/Setting/Service/SettingsService.php

Reply:
Okay fine but this feature is about to provide default values for the default config.xml (in the example from PayPal the configuration is not even defined in a .xml file in the desired directory). 

That said, how should I scan for possible config names? Ofcourse I could open every xml file in the config directory and take a look if it have the correct structure but this kinda slow.

Reply:
ok, maybe we could only consider the default behaviour in the first step. so this is fixed for most of the plugins :+1: 
later we could extend it, so custom config files could also be used

Reply:
Seems legit :)